### ISSUE TITLE
[v6] Incomplete type definitions for MaterialTopTabNavigator

### ISSUE THREAD

**Issue by yfunk** [react-navigation/react-navigation#9769]
### Current behavior

Most likely the same underlying issue as #9485, but that one is incomplete and seems to have gone stale.

```ts
type DummyParamList = {
	home: undefined;
	feed: undefined;
};

const MaterialTopTab = createMaterialTopTabNavigator<DummyParamList>();

export const MaterialTopTabTest = () => {
	return (
		<MaterialTopTab.Navigator tabBar={(props) => <Dummy {...props} />}>
			<MaterialTopTab.Screen name='home' component={Dummy} options={{}} />
		</MaterialTopTab.Navigator>
	);
};
```
When creating and using a typed `MaterialTopTabNavigator` on `@react-navigation/material-top-tabs @6.0.0-next.18` TypeScript / VSCode IntelliSense does not pick up all of the types for `MaterialTopTab.Navigator`. This causes two main issues when working with this library.

### Incomplete IntelliSense / type checking
<img width="763" alt="incomplete-sense" src="https://user-images.githubusercontent.com/14955991/127221936-515f8f47-829a-4c45-9603-1743091ff1d4.png">

Specifically `backBehaviour`, `tabBarPosition`, `lazy`, `lazyPreloadDistance`, `lazyPlaceholder`, `keyboardDismissMode`, `swipeEnabled`, `initialLayout`, `sceneContainerStyle`, `style`, `tabBar` and `tabBarOptions`, which are documented for `6.x (next)`, are missing.

### Implicit `any` at callback props (e.g. `tabBar`)
<img width="763" alt="any-type" src="https://user-images.githubusercontent.com/14955991/127222209-e89178dc-85a8-4015-a8a6-dd2c40f36f50.png">

---

Unfortunately I'm not sure what the exact reason for this issue could be, since the types are all correctly defined at `main/packages/material-top-tabs/src/types.tsx`. According to the date on the issue linked above this issue has been around for some time already.

The problem doesn't occur with any of the other navigators I tested (`Stack` and `BottomTab`) and is reproducible in a fresh React Native 0.64.2 project or in a Expo SDK 42 project (linked below).

### Expected behavior

All props defined for the `MaterialTopTab.Navigator` in `main/packages/material-top-tabs/src/types.tsx` are be picked up by TypeScript and IntelliSense.

### Reproduction

https://github.com/yfunk/material-top-tab-type-issue

### Platform

- [ ] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-bottom-tabs
- [X] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack

### Environment

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               | ^6.0.0-next.17
| @react-navigation/bottom-tabs          | ^6.0.0-next.22
| @react-navigation/material-top-tabs    | ^6.0.0-next.18
| react-native-safe-area-context         | 3.2.0
| react-native-screens                   | ~3.4.0
| react-native-gesture-handler           | ~1.10.2
| react-native-reanimated                | ~2.2.0
| react-native-tab-view                  | ^3.0.1
| react-native-pager-view                | ^5.4.0
| react-native                           | 0.64.2
| expo                                   | SDK 42.0.0
| typescript                                  | ^4.3.5
| node                                   | v15.8.0
| yarn                            | 1.22.5

**Comment by yfunk** [comment]
> 
> 
> Couldn't find version numbers for the following packages in the issue:
> 
>     * `@react-navigation/drawer`
> 
>     * `@react-navigation/material-bottom-tabs`
> 
>     * `@react-navigation/stack`
> 

These packages are not included since they aren't necessary to reproduce the issue

> The versions mentioned in the issue for the following packages differ from the latest versions on npm:
> 
>     * `@react-navigation/native` (found: `6.0.0-next.17`, latest: `5.9.4`)
> 
>     * `@react-navigation/bottom-tabs` (found: `6.0.0-next.22`, latest: `5.11.11`)
> 
>     * `@react-navigation/material-top-tabs` (found: `6.0.0-next.18`, latest: `5.3.15`)
> 
> 
> Can you verify that the issue still exists after upgrading to the latest versions of these packages?

The versions this issue occurs at are `@next`.

**Comment by yfunk** [comment]
I had another look at the code and the whole reason for this issue was an incorrect import over at `react-native-tab-view`, which has [already been fixed](https://github.com/satya164/react-native-tab-view/commit/dad1e49f6cefcd7ba5eb062775f9e9fb06b875ad). 

@satya164 Could you please release the `main` branch of `react-native-tab-view` to include this fix?

**Comment by yfunk** [comment]
Fixed in release [`3.1.0`](https://github.com/satya164/react-native-tab-view/releases/tag/v3.1.0) of `react-native-tab-view`.
